client/session/manager.go

```diff
@@ -0,0 +1,175 @@
+package session
+
+import (
+	"fmt"
+	"net/http"
+	"strings"
+	"sync"
+
+	"github.com/pkg/errors"
+	"golang.org/x/net/context"
+	"google.golang.org/grpc"
+)
+
+// Caller can invoke requests on the session
+type Caller interface {
+	Context() context.Context
+	Supports(method string) bool
```

Comment:
Additionally, it would be more convenient for user code if SupportsMethod takes (serviceName string, method name)  instead of a method url.